cmd/internal/planetscale_edge_database.go

```diff
@@ -314,9 +318,13 @@ func (p PlanetScaleEdgeDatabase) sync(ctx context.Context, tc *psdbconnect.Table
 			}
 		}
 
-		if watchForVgGtidChange && tc.Position != stopPosition {
+		if (watchForVgGtidChange && tc.Position != stopPosition) || (lastRestCount == 0 && len(res.Result) == 0 && tc.Position == stopPosition) {
 			return tc, io.EOF
 		}
+
+		if !(lastRestCount == -1 && len(res.Result) == 0) {
```

Comment:
what's the reason for this guard?

Reply:
In testing, the initial copy phase would trigger the condition above without this, but I don't precisely know why. Experimental!